src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -134,46 +142,76 @@ async function compileTypeScript(config: NodejsDebugConfiguration): Promise<void
 
     // Require tsconfig.json or *.ts in the top-level of the source app, to
     // indicate a typescript Lambda. #2086
-    // Note: we don't use this tsconfig.json for compiling the target=code
-    // Lambda app below, instead we generate a minimal one.
+    // Note: we use this tsconfig.json as a base for compiling the target=code
+    // Lambda app below. If it does not exist, we generate a minimal one.
     const isTsApp = (await findTsOrTsConfig(config.codeRoot, false)) !== undefined
     if (!isTsApp) {
         return
     }
 
-    const buildOutputDir = path.join(config.baseBuildDir, 'output')
-    const buildDirTsFile = await findTsOrTsConfig(buildOutputDir, true)
-    if (!buildDirTsFile) {
-        // Should never happen: `sam build` should have copied the tsconfig.json from the source app dir.
-        throw new Error(`tsconfig.json or *.ts not found in: "${buildOutputDir}/*"`)
-    }
-    // XXX: `sam` may rename the CodeUri (and thus "output/<app>/" dir) if the
-    // original "<app>/" dir contains special chars, so get it this way. #2086
-    const buildDirApp = path.dirname(buildDirTsFile)
-    const buildDirTsConfig = path.join(buildDirApp, 'tsconfig.json')
+    const loadBaseConfig = (tsConfigPath: PathLike) => {
+        if (!existsSync(tsConfigPath)) {
+            return undefined
+        }
 
-    const tsc = await systemutil.SystemUtilities.findTypescriptCompiler()
-    if (!tsc) {
-        throw new Error('TypeScript compiler "tsc" not found in node_modules/ or the system.')
+        try {
+            const tsConfig = JSON.parse(readFileSync(tsConfigPath).toString())
+            getLogger('channel').info(`Using base TypeScript config: ${tsConfigPath}`)
+            return tsConfig
+        } catch (err) {
+            getLogger('channel').error(`Unable to use TypeScript base: ${tsConfigPath}`)
+        }
+
+        return undefined
     }
 
-    // Default config.
-    // Adapted from: https://github.com/aws/aws-toolkit-jetbrains/blob/911c54252d6a4271ee6cacf0ea1023506c4b504a/jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsLambdaBuilder.kt#L60
-    const defaultTsconfig = {
-        compilerOptions: {
+    const tsConfigPath = path.join(config.codeRoot, tsConfigFile)
```

Comment:
It'd be really nice if we could write this to the build directory if possible. Maybe we can use [extends](https://www.typescriptlang.org/tsconfig#extends) instead of directly merging config files?